### PR TITLE
Lazy loading of images

### LINKED ISSUE #433: UI enhancements
There are many places in the application where the UI breaks down.

* The home page is nice and responsive

* But the feed page is the place where the design is not responsive enough.

* The UI breaks down and the app feels less sophisticated.

* This is the umbrella issue for the UI enhancement in feed component

* [ ]  Fix the responsiveness of feed cards

* [ ] Make separate button to show statistics on mobile, Make the info box UI better

* [ ] The Image and video grids can be improved in design, Not including borders, and gracefully handling the `404` error for them

* [ ] Improving the design of the header

### LINKED ISSUE #471: Add lazy loading of images in the various parts of the application
**Parent Issue #433**
* Use Asynchronous image loading on demand.

### PR DESCRIPTION
**Pull request by hemantjadon**
* Lazy loading image depending on the position of the element wrt to viewport

* Using Intersection obsever to detect

* Using fetchAPI to asynchronoulsy fetch the Binary data stream and convert it into image

* Add semantic loading information

* Along with the error messages specially 404 errors for images of varied sizes.

https://hemantjadon.github.io/loklak_search

**Closes #471**
**In reference to #433**

### CODE REVIEW

**Comment by hemantjadon** [comment]
@Achint08 @singhpratyush @vibhcool @kavithaenair @djmgit @SKrPl  open for review

**Comment by Achint08** [comment]
@hemantjadon Test link please.

**Comment by hemantjadon** [comment]
@Achint08 @singhpratyush @vibhcool @kavithaenair @djmgit Please review

**Comment by vibhcool** [comment]
the link isn't working 😅

File: src/app/lazy-img/lazy-img.component.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	Output,
+	EventEmitter,
+	ElementRef,
+	ChangeDetectorRef,
+	OnInit,
+	OnDestroy
+} from '@angular/core';
+import { LazyImgService } from './lazy-img.service';
+
+
```
Comment (vibhcool) [inline src/app/lazy-img/lazy-img.component.ts]: extra line

**Comment by hemantjadon** [comment]
@vibhcool the link seems to be working. Please check again. Do activate the load unsafe scripts from the url bar. see networks panel for to see the requests actually getting deferred. Use chrome latest or firefox latest I still have to polyfill this

**Comment by hemantjadon** [comment]
@Achint08 @vibhcool Polyfilled the Intersection observer now check

**Comment by Achint08** [comment]
Please update the test link. @hemantjadon

**Comment by hemantjadon** [comment]
@Achint08 done 👍

**Pull request merged**